<launch>
    <!-- Start the ingestor node -->
    <!-- Note: type matches the executable name from CMakeLists.txt -->
    <node pkg="timescaledb_ingestor" type="timescaledb_ingestor_cpp_node" name="timescaledb_ingestor_node" output="screen">
        <!-- Load parameters FROM the YAML file INTO the node's private namespace -->
        <!-- Ensure config/ingest_params.yaml has the correct parameters -->
        <rosparam command="load" file="$(find timescaledb_ingestor)/config/ingest_params.yaml" />

    </node>
</launch>